contracts/entry-point/tests/test_reply_messages.rs

```diff
@@ -0,0 +1,227 @@
+use skip_api_entry_point::state::SWAP_AND_ACTION_REQUEST_TEMP_STORAGE;
+
+pub struct Params {
+    pub funds: Vec<Coin>,
+    pub reply: Reply,
+    pub expected_error_string: String,
+    pub storage: Option<SwapActionTempStorage>,
+    pub expected_messages: Vec<SubMsg>,
+}
+
+//  Helper function to test all replies
+pub fn test_reply(params: Params) -> ContractResult<()> {
+    // Create mock dependencies
+    let mut deps = mock_dependencies_with_balances(&[(
+        "entry_point",
+        &[Coin::new(1_000_000, "osmo"), Coin::new(1_000_000, "untrn")],
+    )]);
+
+    // Create mock env
+    let env = mock_env();
+
+    // Update storage
+    if let Some(swap_action) = params.storage.clone() {
+        SWAP_AND_ACTION_REQUEST_TEMP_STORAGE.save(deps.as_mut().storage, &swap_action)?;
+    }
+
+    // Call reply with the given test parameters
+    let res = skip_api_entry_point::contract::reply(deps.as_mut(), env, params.reply);
+
+    // Assert the behavior is correct
+    match res {
+        Ok(res) => {
+            // Assert the test did not expect an error
+            assert!(
+                params.expected_error_string.is_empty(),
+                "expected test to error with {:?}, but it succeeded",
+                params.expected_error_string
+            );
+
+            assert_eq!(res.messages, params.expected_messages);
+
+            // Verify the in progress swap and action message was removed from storage
+            match SWAP_AND_ACTION_REQUEST_TEMP_STORAGE.load(&deps.storage) {
+                Ok(swap_and_action_request) => {
+                    panic!(
+                        "expected in progress ibc transfer to be removed: {:?}",
+                        swap_and_action_request
+                    )
+                }
+                Err(err) => assert_eq!(
+                    err,
+                    StdError::NotFound {
+                        kind: "skip_api_entry_point::reply::SwapActionTempStorage".to_string(),
+                    }
+                ),
+            };
+        }
+        Err(err) => {
+            // Assert the test expected an error
+            assert!(
+                !params.expected_error_string.is_empty(),
+                "expected test to succeed, but it errored with {:?}",
+                err
+            );
+
+            // Assert the error is correct
+            assert_eq!(err.to_string(), params.expected_error_string);
+        }
+    }
+
+    Ok(())
+}
+
+#[test]
+pub fn verify_funds_sent_on_out_of_gas_error() {
+    let recovery_addr = Addr::unchecked("recovery_addr").to_string();
+    let bank_msg = BankMsg::Send {
+        to_address: recovery_addr,
+        amount: vec![Coin::new(1_000_000, "osmo")],
+    };
+
+    let sub_msg = SubMsg::new(CosmosMsg::Bank(bank_msg));
+
+    let params = Params {
+        funds: vec![],
+        reply: Reply {
+            id: 1,
+            result: SubMsgResult::Err("Out of gas".to_string()),
```

Comment:
I don't think this matters much, but in reality we would not be able to send bank the funds on out of gas error because an out of gas error gets handled immediately back at the sdk level not the cosmwasm vm level, so this test may give a non-cosmwasm/cosmos-sdk native person the wrong impression that we can recover in those cases

Reply:
Ahh good point, updated to slippage tolerance exceeded